### PR TITLE
[dyno] add new port

### PR DESCRIPTION
**Pull request by jiayuehua**
- [X] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [X] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [X] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html)
- [X] The versioning scheme in `vcpkg.json` matches what upstream says.
- [X] The license declaration in `vcpkg.json` matches what upstream says.
- [X] The installed as the "copyright" file matches what upstream says.
- [X] The source code of the component installed comes from an authoritative source.
- [X] The generated "usage text" is accurate. See [adding-usage](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/examples/adding-usage.md) for context.
- [X] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [X] Only one version is in the new port's versions file.
- [X] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/dyno/fix-deps.patch [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
++++ b/CMakeLists.txt
+@@ -14,9 +14,8 @@ add_library(dyno INTERFACE)
+ add_library(Dyno::dyno ALIAS dyno)
+ target_compile_features(dyno INTERFACE cxx_std_17)
+ target_include_directories(dyno INTERFACE "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>")
+-find_package(Hana REQUIRED)
+-find_package(CallableTraits REQUIRED)
+-target_link_libraries(dyno INTERFACE hana callable_traits)
++find_package(Boost REQUIRED)
++target_link_libraries(dyno INTERFACE Boost::boost)
```
Comment (dg0yt) [inline ports/dyno/fix-deps.patch:13]: Should this be used with the actual components?

Comment (jiayuehua) [reply ports/dyno/fix-deps.patch:13]: Those two are header only, if changed to 

`  find_package(Boost REQUIRED COMPONENTS hana)
  target_link_libraries(mp_test PRIVATE Boost::boost)`

then report error 
``
1> [CMake] CMake Warning at /usr/local/share/cmake-3.27/Modules/FindBoost.cmake:2233 (message):
1> [CMake]   No header defined for hana; skipping header check (note: header-only
1> [CMake]   libraries have no designated component)
1> [CMake] Call Stack (most recent call first):
1> [CMake]   /home/rob/opensrc/vcpkg/installed/x64-linux/share/boost/vcpkg-cmake-wrapper.cmake:11 (_find_package)
1> [CMake]   /home/rob/opensrc/vcpkg/scripts/buildsystems/vcpkg.cmake:813 (include)
1> [CMake]   CMakeLists.txt:32 (find_package)
1> [CMake] 
1> [CMake] 
1> [CMake] CMake Error at /usr/local/share/cmake-3.27/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
1> [CMake]   Could NOT find Boost (missing: hana) (found version "1.83.0")

Comment (dg0yt) [reply ports/dyno/fix-deps.patch:13]: Okay, understood. Thank you!

**Comment by jiayuehua** [comment]
@LilyWangLL Would you please review this pr?

**Comment by LilyWangLL** [comment]
Usage test passed on `x64-windows`.

**Review by LilyWangLL — DISMISSED** [review]

**Review by data-queue — DISMISSED** [review]

**Comment by data-queue** [comment]
Can you fix the merge conflict?

**Comment by data-queue** [comment]
Thanks for the PR!

**Pull request merged**